enclave-runtime/src/tests.rs

```diff
@@ -222,38 +246,50 @@ fn test_differentiate_getter_and_call_works() {
 fn test_create_block_and_confirmation_works() {
 	// given
 	let (rpc_author, _, shard, mrenclave, shielding_key, state_handler) = test_setup();
-	let stf_executor = StfExecutor::new(Arc::new(OcallApi), state_handler.clone());
+	let stf_executor = Arc::new(StfExecutor::new(Arc::new(OcallApi), state_handler.clone()));
+	let top_pool_executor = TopPoolOperationExecutor::<Block, SignedBlock, _, _>::new(
+		rpc_author.clone(),
+		stf_executor.clone(),
+	);
+	let block_composer = BlockComposer::<Block, SignedBlock, _, _, _, _>::new(
+		test_account(),
+		state_key(),
+		rpc_author.clone(),
+		stf_executor,
+	);
```

Comment:
in some of the tests here we now need the top pool executor and the block composer explicitly